usr/src/cmd/acct/acctprc.c

```diff
@@ -156,7 +156,7 @@ void print_node(const void *node, VISIT order, int level) {
 
 	if (order == postorder || order == leaf) {
 		tb.ta_uid = (*(struct utab **)node)->ut_uid;
-		CPYN(tb.ta_name, (char *)uidtonam((*(struct utab **)node)->ut_uid));
```

Comment:
This is probably another case where what's really going on is the an implicit declaration of `uidtonam` leading the compiler to think it returns `int`.  You don't need to cast through to the type it returns, but fix the implicit declaration.

You can probably catch a lot here by removing any `-Wno-implicit-function-declaration` from the makefiles.

Reply:
Yeah its right in others but you are right missing a definition for uidtonam() added, fixed.

Reply:
There's a _lot_ of implicit declaration warnings in here if you remove the entry from the Makefile to suppress them.
We should probably fix them all and switch to ANSI C to be extra sure everything is ok.  Without doing this, the compiler knows nothing about the types of parameters or returns and so can't tell us what's up

Reply:
I will give it a try to see what it really needs this is really old and crufty code.

Reply:
Hmm this is really a can of worms probably better to just drop this patch and one day either rwrite it from scratch or try fixing it if that is possible get the feeling this is far from 64 bits compatible.